api/test/service/store.test.js

```diff
@@ -97,10 +98,40 @@ test('store/add returns signed url for uploading', async (t) => {
     with: spaceDid,
     link,
     headers: {
+      'content-length': size,
       'x-amz-checksum-sha256': base64pad.baseEncode(link.multihash.digest)
     }
   })
-  t.is(storeAdd.url && new URL(storeAdd.url).pathname, `/${link}/${link}.car`)
+  const url = storeAdd.url && new URL(storeAdd.url)
+  if (!url) {
+    throw new Error('Expected presigned url in response')
+  }
+
+  const signedHeaders = url.searchParams.get('X-Amz-SignedHeaders')
+  t.is(signedHeaders, 'content-length;host;x-amz-checksum-sha256', 'content-length and checksum must be part of the signature')
+
+  const key = url?.pathname.slice(11) // minio puts bucket name at start of path
+  t.is(key, `/${link}/${link}.car`)
+
+  const goodPut = await fetch(url, {
+    method: 'PUT',
+    mode: 'cors',
+    body: data,
+    headers: storeAdd.headers
+  })
+  t.is(goodPut.status, 200, await goodPut.text())
+
+  const unexpectedLength = await fetch(url, {
+    method: 'PUT',
+    mode: 'cors',
+    body: new Uint8Array([11, 22, 34, 44, 55, 1]), // one byte more
+    headers: {
+      ...storeAdd.headers,
+      'content-length': '6'
+    }
+  })
+  // expect signature to not match
+  t.is(unexpectedLength.status, 403, await unexpectedLength.text())
```

Comment:
Test description does not match this flow. Can we change it or decouple in 2 tests?

Reply:
strictly this *is* a "signature not match" failure as the content-length header that was used to sign the url has the value 5, and when using it we send 6, so it fails when validating the url sig.

Reply:
i've split the tests anyway